x-pack/plugins/observability/public/data_handler.ts

```diff
@@ -28,9 +28,26 @@ export function getDataHandler<T extends ObservabilityApp>(appName: T) {
   }
 }
 
-export async function fetchHasData() {
+function getPromiseResult(promiseResult: PromiseFulfilledResult<any> | PromiseRejectedResult) {
```

Comment:
Shouldn't these always return a boolean?
```suggestion
function getPromiseResult(promiseResult: PromiseSettledResult<boolean>) {
```